Controller/ProductController.cs

```diff
@@ -0,0 +1,349 @@
+                {
+                    product = new Product() { Id = Guid.NewGuid(), Name = name, Price = price, Quantity = quantity };
+                    _productService.AddProduct(product);
+                    Console.WriteLine("Product added successfully.");
+                }
+                else
+                {
+                    Console.WriteLine("Product already exists, cannot be added");
+                }
+                Console.WriteLine("\nPress any key to return to menu.");
+                Console.ReadKey();
+                Console.Clear();
+            }
+            catch(Exception e)
+            {
+                Console.WriteLine("Something went wrong.");
+                CustomLogger.Logger.LogError(e);
+            }
+        }
+
+        public void ShowAllProductsCustomer()
+        {
+            try
+            {
+                var products = _productService.GetAllProducts().Where(prod => prod.Quantity > 0).ToList();
+                if (products.Count <= 0)
+                {
+                    Console.WriteLine("No products available.");
+                    Console.WriteLine("\nPress any key to return to menu.");
+                    Console.ReadKey();
+                    Console.Clear();
+                    return;
+                }
+                int i = 1;
+                foreach (var product in products)
+                {
+                    Console.Write($"Sr. No. {i++},  ");
+                    product.DisplayInfo();
+                }
+                Console.WriteLine("\nPress any key to return to menu.");
+                Console.ReadKey();
+                Console.Clear();
+            }
+            catch(Exception e)
+            {
+                Console.WriteLine("Something went wrong.");
+                CustomLogger.Logger.LogError(e);
+            }
+        }
+
+        public void ShowAllProductsAdmin()
+        {
+            try
+            {
+                var products = _productService.GetAllProducts().ToList();
+                if (products.Count <= 0)
+                {
+                    Console.WriteLine("No products available.");
+                    Console.WriteLine("\nPress any key to return to menu.");
+                    Console.ReadKey();
+                    Console.Clear();
+                    return;
+                }
+                int i = 1;
+                foreach (var product in products)
+                {
+                    Console.Write($"Sr. No. {i++},  ");
+                    product.DisplayInfo();
+                }
+                Console.WriteLine("\nPress any key to return to menu.");
+                Console.ReadKey();
+                Console.Clear();
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("Something went wrong.");
+                CustomLogger.Logger.LogError(e);
+            }
+        }
+
+        public void UpdateProductDetails()
```

Comment:
This is a huge function, try to refactor it and have small functions handling single responsibility.